diff --git a/ipv8/REST/root_endpoint.py b/ipv8/REST/root_endpoint.py
index 2fca7c9..9b42d9b 100644
--- a/ipv8/REST/root_endpoint.py
+++ b/ipv8/REST/root_endpoint.py
@@ -1,6 +1,14 @@
 from __future__ import absolute_import
 
+import os
 from twisted.web import resource
+from twisted.web.static import File
+import tlswitness
+from tlswitness.defs import BASE_DIR, PROVIDERS
+from tlswitness.provider.digid import DigidProvider
+from tlswitness.provider.github import GithubProvider
+from tlswitness.provider.mijnoverheid import MijnOverheidProvider
+from tlswitness.restapi.root_endpoint import APIEndpoint
 
 from .attestation_endpoint import AttestationEndpoint
 from .dht_endpoint import DHTEndpoint
@@ -29,3 +37,13 @@ class RootEndpoint(resource.Resource):
         self.putChild(b'overlays', OverlaysEndpoint(session))
         self.putChild(b'dht', DHTEndpoint(session))
         self.putChild(b'tunnel', TunnelEndpoint(session))
+
+        cert_fn = os.path.join(BASE_DIR, 'cacert.pem')
+        if os.path.exists(cert_fn):
+            os.environ['SSL_CERT_FILE'] = cert_fn
+
+        vars = globals()
+        proxy_args = ('https://tlswitness.org:8123', os.path.join(BASE_DIR, 'cert_web.crt'))
+        providers = {name: vars[settings['class']](*proxy_args) for name, settings in PROVIDERS.iteritems()}
+        self.putChild(b'api', APIEndpoint(providers, 'http://localhost:8085'))
+        self.putChild(b'gui', File(os.path.join(BASE_DIR, 'webapp', 'dist')))
